src/blocks/privacy.rs

```diff
@@ -0,0 +1,212 @@
+    )?;
+    let mut format_alt = config.format_alt.with_default("{ $icon_audio $info_audio |}{ $icon_audio_sink $info_audio_sink |}{ $icon_video $info_video |}{ $icon_webcam $info_webcam |}{ $icon_unknown $info_unknown |}")?;
+
+    let mut drivers: Vec<Box<dyn PrivacyMonitor + Send + Sync>> = Vec::new();
+
+    for driver in &config.driver {
+        drivers.push(match driver {
+            #[cfg(feature = "pipewire")]
+            PrivacyDriver::Pipewire(driver_config) => {
+                Box::new(pipewire::Monitor::new(driver_config).await?)
+            }
+            PrivacyDriver::V4l(driver_config) => {
+                Box::new(v4l::Monitor::new(driver_config, api.error_interval).await?)
+            }
+        });
+    }
+
+    loop {
+        let mut widget = Widget::new().with_format(format.clone());
+
+        let mut info = PrivacyInfo::new();
+        //Merge driver info
+        for driver_info in try_join_all(drivers.iter_mut().map(|driver| driver.get_info())).await? {
+            for (type_, mapping) in driver_info {
+                let existing_mapping = info.entry(type_).or_default();
+                for (source, dest) in mapping {
+                    existing_mapping.entry(source).or_default().extend(dest);
+                }
+            }
+        }
+        if !info.is_empty() {
+            widget.state = State::Warning;
+        }
+
+        let mut values = Values::new();
+
+        if let Some(info_by_type) = info.get(&Type::Audio) {
+            values.extend(map! {
+                "icon_audio" => Value::icon("microphone"),
+                "info_audio" => Value::text(format!("{:?}", info_by_type))
+            });
+        }
+        if let Some(info_by_type) = info.get(&Type::AudioSink) {
+            values.extend(map! {
+                "icon_audio_sink" => Value::icon("volume"),
+                "info_audio_sink" => Value::text(format!("{:?}", info_by_type))
+            });
+        }
+        if let Some(info_by_type) = info.get(&Type::Video) {
+            values.extend(map! {
+                "icon_video" => Value::icon("xrandr"),
+                "info_video" => Value::text(format!("{:?}", info_by_type))
+            });
+        }
+        if let Some(info_by_type) = info.get(&Type::Webcam) {
+            values.extend(map! {
+                "icon_webcam" => Value::icon("webcam"),
+                "info_webcam" => Value::text(format!("{:?}", info_by_type))
+            });
+        }
+        if let Some(info_by_type) = info.get(&Type::Unknown) {
+            values.extend(map! {
+                "icon_unknown" => Value::icon("unknown"),
+                "info_unknown" => Value::text(format!("{:?}", info_by_type))
+            });
+        }
+
+        widget.set_values(values);
+
+        api.set_widget(widget)?;
+
+        select! {
+            _ = api.wait_for_update_request() => (),
+            _ = select_all(drivers.iter_mut().map(|driver| driver.wait_for_change())) =>(),
```

Comment:
Do you think it is worth it to keep track which driver has an update and only poll this specific driver?

Reply:
I don't think that it really matters for these two drivers. `v4l`'s `get_info` is just reading from `/proc` and `pipewire`'s `get_info` is just querying the list of nodes/links that are already discovered. Since we don't have to do any expensive I/O, I think it's ok as-is.

Reply:
Alright, lets implement this when/if we have more drivers.